src/main/java/com/example/demo/service/PostService.java

```diff
@@ -0,0 +1,59 @@
+package com.example.demo.service;
+
+import java.util.Objects;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.example.demo.dto.post.PostCreateRequestDto;
+import com.example.demo.dto.post.PostFindResponseDto;
+import com.example.demo.dto.post.PostsFindResponseDto;
+import com.example.demo.model.member.Member;
+import com.example.demo.model.member.Social;
+import com.example.demo.model.post.Genre;
+import com.example.demo.model.post.Post;
+import com.example.demo.repository.MemberRepository;
+import com.example.demo.repository.PostRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class PostService {
+
+	private final PostRepository postRepository;
+	private final MemberRepository memberRepository;
+
+	@Transactional
+	public Long createPost(PostCreateRequestDto postRequestDto) {
+		//TODO : 임의 유저 생성, 소셜 로그인 구현 후 삭제
+		Member member = new Member("url", "name", 0, 1,
+			10, 10, "token", Social.GOOGLE, "social");
+		memberRepository.save(member);
+
+		Post post = postRequestDto.toEntity(member);
+		postRepository.save(post);
+
+		return post.getId();
+	}
+
+	public PostsFindResponseDto findAllPosts(Genre genre, Boolean possible) {
```

Comment:
로직을 한눈에 알아보기가 어려운거 같애서

```suggestion
	public PostsFindResponseDto findAllPosts(Genre genre, Boolean possible) {
        switch (ScenarioBranch.generateScenario(genre, possible)) {
			case BOTH_NOT_NULL -> {
				// 로직 넣기
			}
			case GENRE_ONLY_NOT_NULL -> {

			}
			case POSSIBLE_ONLY_NOT_NULL -> {
			}
			case BOTH_NULL -> {
			}
		}
		return Genre.K_POP;
    }

    private enum ScenarioBranch {
		BOTH_NOT_NULL,
		GENRE_ONLY_NOT_NULL,
		POSSIBLE_ONLY_NOT_NULL,
		BOTH_NULL;

		static ScenarioBranch generateScenario(Optional<Genre> genre, Optional<Boolean> possible) {
			if (genre.isPresent() && possible.isPresent()) {
				return BOTH_NOT_NULL;
			} else if (genre.isPresent()) {
				return GENRE_ONLY_NOT_NULL;
			} else if (possible.isPresent()) {
				return POSSIBLE_ONLY_NOT_NULL;
			} else {
				return BOTH_NULL;
			}
		}
	}
```
이런 느낌으로 switch-case를 이용하는 방법은 어떤가요??

Reply:
Optional을 넣은 이유는 이전에 controller단에서 만약 Optional로 처리한다면 이런 느낌일 것 같다는 생각으로 넣었습니다.

Reply:
파라미터 2개를 switch로 어떻게 나누나 고민했었는데 감사합니다 :) 좋은 방식인 것 같아요!
다만 nonNull로 충분히 표현할 수 있는데 Optional로 감싸서 오는 이점이 있는 지는 아직 잘 와닿지 않는 것 같아요 🤔